README.md

```diff
@@ -100,44 +103,68 @@ t := myQueue.NewTask(&taskq.TaskOptions{
 })
 
 // Say "Hello World".
-t.Call("World")
+myQueue.Add(t.WithArgs(context.Background(), "World"))
 
 // Same using Message API.
-t.AddMessage(taskq.NewMessage("World"))
+myQueue.Add(t.WithMessage(taskq.NewMessage("World")))
 
 // Say "Hello World" with 1 hour delay.
 msg := taskq.NewMessage("World")
 msg.Delay = time.Hour
-t.AddMessage(msg)
+myQueue.Add(msg)
 
 // Say "Hello World" once.
 for i := 0; i < 100; i++ {
     msg := taskq.NewMessage("hello")
     msg.Name = "hello-world" // unique
-    t.Add(msg)
+    myQueue.Add(msg)
 }
 
 // Say "Hello World" once with 1 hour delay.
 for i := 0; i < 100; i++ {
     msg := taskq.NewMessage("hello")
     msg.Name = "hello-world"
     msg.Delay = time.Hour
-    t.Add(msg)
+    myQueue.Add(msg)
 }
 
 // Say "Hello World" once in an hour.
 for i := 0; i < 100; i++ {
-    t.CallOnce(time.Hour, "hello")
+    myQueue.Add(t.OnceWithArgs(time.Hour, "hello"))
 }
 
 // Say "Hello World" for Europe region once in an hour.
 for i := 0; i < 100; i++ {
     msg := taskq.NewMessage("hello")
     msg.OnceWithArgs(time.Hour, "europe") // set delay and autogenerate message name
-    t.Add(msg)
+    myQueue.Add(msg)
 }
 ```
 
+## Message deduplication
+If a `Message` has a `Name` then this will be used as unique identifier and messages with the same name will be deduplicated (i.e. not processed again) within a 24 hour period (or possibly longer if not evicted from local cache after that period). Where `Name` is omitted then non deduplication occurs and each message will be processed. `Task`'s `WithMessage` and `WithArgs` both produces messages with no `Name` so will not be deduplicated. `OnceWithArgs` sets a name based off a consistent hash of the arguments and a quantised period of time (i.e. 'this hour', 'today') passed to `OnceWithArgs` a `period`. This guarantees that the same function will not be called with the same arguments during `period'.
+
+## Contextual handlers
+If a task is registered with a handler that takes a Go `context.Context` as its first argument then when that handler is invoked it will be passed the same `Context` that was based to `Consumer.Start(ctx)`. This can be used to transmit a signal to abort to all tasks being processed:
```

Comment:
One question I have about the API is that `WithMessage` takes a context that it passed to `newMessage`. Is there actually any point in this? I can't see how this context ever gets used since we serialise and deserialise the message before it ever reaches the handler. And before then we set the context to the value held by the worker.

Unless I'm missing something it seems like this per-message context could/should be removed?